vespalib/src/vespa/vespalib/util/alloc.cpp

```diff
@@ -465,6 +466,15 @@ Alloc::allocMMap(size_t sz)
     return Alloc(&MMapAllocator::getDefault(), sz);
 }
 
+Alloc
+Alloc::alloc()
+{
+    if (_G_defaultAutAllocator == nullptr) {
+        _G_defaultAutAllocator = &AutoAllocator::getDefault();
+    }
+    return Alloc(_G_defaultAutAllocator);
```

Comment:
If this method can be called from multiple threads when `_G_defaultAutAllocator == nullptr`, this technically triggers a data race. If so, consider replacing with
```c++
std::atomic<alloc::MemoryAllocator*> _G_defaultAutAllocator = nullptr;
...
if (_G_defaultAutAllocator.load(std::memory_order_relaxed) == nullptr) {
        _G_defaultAutAllocator.store(&AutoAllocator::getDefault(), std::memory_order_relaxed);
    }
    return Alloc(_G_defaultAutAllocator.load(std::memory_order_relaxed));
```
Since `&AutoAllocator::getDefault()` is created at process init time there is no need for any barriers, but still need well defined load/stores to the pointer itself.